### PR TITLE
[cluster-agent] Fix CWS Instrumentation webhook label selector and error handling

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR tackles 4 issues:
- Pod Label selection doesn't work OOTB for `pods/exec` admission selection - this PR implements a check on "admission.datadoghq.com/cws-instrumentation.enabled" manually in the `pods/exec` admission webhook.
- During [a previous refacto](https://github.com/DataDog/datadog-agent/pull/23770/files#diff-286f4f17fe0530a3b612c38ed0b7081903bb8ca50e40852122a6228f4dc33833R501-R510), the `pods/exec` mutating webhook function for "cws-instrumentation" was mistakenly called twice. This PR removes the unnecessary second call.
- Error handling during the remote copy was flaky, this PR improves it to make sure that only successful remote copies return no error.
- By default, `cws-instrumentation` was copied at the root of the filesystem, but in some cases, the container default user doesn't have write access on `/`, which is why the default is now to write to `/tmp` (but it is configurable).

### Motivation

This fixes bring us one step closer to enabling this feature by default.

### Describe how to test/QA your changes

See https://github.com/DataDog/datadog-agent/pull/24175

### CODE REVIEW

File: pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go [hunk 1]
```diff
@@ -477,7 +487,7 @@ func (ci *CWSInstrumentation) injectCWSCommandInstrumentation(exec *corev1.PodEx
 				metrics.CWSExecInstrumentationAttempts.Observe(1, ci.mode.String(), "false", cwsPodNotInstrumentedReason)
 				return false, nil
 			}
-			cwsInstrumentationRemotePath = filepath.Join(cwsMountPath, "cws-instrumentation")
+			cwsInstrumentationRemotePath = filepath.Join(cwsMountPath, ci.directoryForRemoteCopy, "cws-instrumentation")
```
Comment (wdhif) [inline pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go:490]: nit: The `cwsInstrumentationRemotePath` could be done outside the `if/else` with just the `cwsMountPath` being set by the `if/else`.

**Review by wdhif — APPROVED** [review]

**Review by dustmop — APPROVED** [review]
LGTM for agent-shared-components

File: pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go (hunk 1 above)
Comment (Gui774ume) [reply pkg/clusteragent/admission/mutate/cwsinstrumentation/cws_instrumentation.go]: Done 👍🏻

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**